src/openrct2/ride/RideData.cpp

```diff
@@ -454,3 +454,15 @@ TrackDrawerEntry getTrackDrawerEntry(const RideTypeDescriptor& rtd, bool isInver
 
     return descriptor.Regular;
 }
+
+int32_t RideTypeDescriptor::GetLegacyBoosterSpeed(int32_t unifiedSpeed) const
+{
+    return unifiedSpeed * 2 / LegacyBoosterSettings.BoosterSpeedFactor;
```

Comment:
I don’t understand this code. Say `LegacyBoosterSettings.BoosterSpeedFactor` is 1 (representing 1/2 speed), then it will _multiply_ by 2. I would expect the opposite.

Reply:
GetLegacy and GetUnified are inverses of each other. You supply GetUnified with a legacy speed to get a unified speed, and you supply GetLegacy with a unified speed to get a legacy speed.

GetUnified is the function that is called to convert the TrackElement's BrakeBoosterSpeed into something the vehicle can use. In the full Unified Brake Speed PR the GetLegacy function is used to convert a Unified Speed (BrakeBoosterSpeed of TrackElement owned by the track element's) into a Legacy Speed (dropping precision to vanilla levels in the process) then back to a Unified Speed (based on the vehicle's RTD). This process only applies if the vehicle has the legacy flag enabled.

In this instance, I could possibly multiply the two BoosterSpeedFactor constants together then mask the output by the vehicle's RTD's BoosterSpeedFactor. I just thought of this so I don't know which way is better - if this is better then I'll just drop the GetLegacy function.

Reply:
I've visited this idea and I've determined that the two users can be decreased to one by "inlining" the function into the equation, leaving it with one user, so I just inlined it there too.